src/vellum/workflows/runner/runner.py

```diff
@@ -263,17 +263,46 @@ def _emit_event(self, event: WorkflowEvent) -> WorkflowEvent:
         return event
 
     def _run_work_item(self, node: BaseNode[StateType], span_id: UUID) -> None:
-        execution = get_execution_context()
-        self._workflow_event_inner_queue.put(
-            NodeExecutionInitiatedEvent(
-                trace_id=execution.trace_id,
-                span_id=span_id,
-                body=NodeExecutionInitiatedBody(
-                    node_definition=node.__class__,
-                    inputs=node._inputs,
-                ),
-                parent=execution.parent_context,
-            )
+        for event in self.run_node(node, span_id):
+            self._workflow_event_inner_queue.put(event)
+
+    def run_node(
+        self,
+        node: "BaseNode[StateType]",
+        span_id: UUID,
+        execution_context: Optional["ExecutionContext"] = None,
+    ) -> Generator["WorkflowEvent", None, None]:
```

Comment:
This should be a generator of the four node events. We should have a union of that handy, but if not, make one